### PR TITLE
Moved DAL to mongo-db

### PR DESCRIPTION
**Pull request by vikrud**
Rewritten repositories in mongoose lib

### CODE REVIEW

File: modules/user/user.repository.js [hunk 1]
```diff
@@ -1,39 +1,60 @@
...
+const userAggregateForm = {
+    _id: 0,
+    id: 1,
+    name: 1,
+    surname: 1,
+    email: 1,
+    phone: 1,
+};
+
+mongoose.connect(url, {
```
Comment (rudenkose) [inline modules/user/user.repository.js]: move this code to index.js 

- [x] move DB connection to index.js

File: modules/user/user.repository.js [hunk 2]
```diff
@@ -1,39 +1,60 @@
...
+        const mongoId = user._id;
+
+        return mongoId;
     }
 
-    async readUserById(id) {
-        const usersArr = await this.readAllUsers();
-        const index = await this.findUserIndexById(id);
+    async readUserById(userId) {
+        const mongoId = await this.findMongoIdByUserId(userId);
```
Comment (rudenkose) [inline modules/user/user.repository.js]: that looks overhead to have two queries to DB here. Let's try get user with one query

- [x] get user by id with one query

File: modules/user/user.repository.js [hunk 3]
```diff
@@ -45,44 +66,36 @@ class UserRepository {
...
-        if (isSaved) {
-            throw new Error(500);
-        }
+        const mongoId = await this.findMongoIdByUserId(id);
+        const deletedUser = await UserModel.findByIdAndDelete(mongoId, {
+            select: userAggregateForm,
+        });
 
-        return true;
+        return [deletedUser];
```
Comment (rudenkose) [inline modules/user/user.repository.js]: It's better to return nothing here.

File: modules/user/user.repository.js [hunk 4]
```diff
@@ -45,44 +66,36 @@ class UserRepository {
...
+    async updateUser(updatedUserData) {
+        mongoose.set("useFindAndModify", false);
+        const mongoId = await this.findMongoIdByUserId(updatedUserData.id);
+        const updatedUser = await UserModel.findByIdAndUpdate(
+            mongoId,
+            updatedUserData,
+            { new: true, select: userAggregateForm }
+        );
 
         return [updatedUser];
```
Comment (rudenkose) [inline modules/user/user.repository.js]: It's better to return nothing here.

File: modules/user/user.repository.js [hunk 5]
```diff
@@ -45,44 +66,36 @@ class UserRepository {
...
 
-    async updateUser(updatedUser) {
-        const usersArr = await this.readAllUsers();
-        const userIndex = await this.findUserIndexById(updatedUser.id);
-
-        usersArr[userIndex] = updatedUser;
-
-        await this.saveFileUsers(usersArr);
+    async updateUser(updatedUserData) {
+        mongoose.set("useFindAndModify", false);
```
Comment (rudenkose) [inline modules/user/user.repository.js]: I am not sure, that this code relates to update user logic. This looks like mongoose configuration, are you sure that this code should be here?

Comment (vikrud) [reply modules/user/user.repository.js]: I will transfer this code to the Mongoose connection configuration
`mongoose.connect(url, {
    useUnifiedTopology: true,
    useNewUrlParser: true,
    useFindAndModify: false,
});`

**Review by rudenkose — APPROVED** [review]

**Pull request merged**